### PR TITLE
Update Mongodb to latest driver

### PR DESCRIPTION
**Pull request by nturinski**
Fixes https://github.com/microsoft/vscode-cosmosdb/issues/1197 
Fixes https://github.com/microsoft/vscode-cosmosdb/issues/1201

Some complications arose from updating Mongodb's type definitions which led me to have to update node's type defs which is why I had to make some changes to the Graph code.

I did some very basic tests for Graph (made sure I could still create an account, db, graph, and that the HTML view still works)

But connecting to dbs work and also works through the App Service Extension via the "Connect to Cosmos DB" node.

Edit: Builds are still failing on CI for some reason, but I'll figure that out on Monday 😓

### CODE REVIEW

File: src/mongo/mongoConnectionStrings.ts [hunk 1]
```diff
@@ -48,18 +49,18 @@ export async function parseMongoConnectionString(connectionString: string): Prom
...
         // the replicaset host name, which is different than what is in the connection string
         // "s" is not part of ReplSet static definition but can't find any official documentation on it. Yet it is definitely there at runtime. Grandfathering in.
         // tslint:disable-next-line:no-any
         let rs: any = serverConfig;
-        host = rs.s.replset.s.seedlist[0].host;
-        port = rs.s.replset.s.seedlist[0].port;
+        host = rs.s.options.servers[0].host;
+        port = rs.s.options.servers[0].port;
     } else {
         host = serverConfig['host'];
```
Comment (EricJizbaMSFT) [inline src/mongo/mongoConnectionStrings.ts]: So the top case changed - I wonder if the else case also changed? That would be bad if we only work in the Azure case and break other mongo dbs

Comment (nturinski) [reply src/mongo/mongoConnectionStrings.ts]: Do you just test this with a local mongodb instance running?

Comment (EricJizbaMSFT) [reply src/mongo/mongoConnectionStrings.ts]: I haven't tested Cosmos in a while haha... but that sounds like a good option

Comment (nturinski) [reply src/mongo/mongoConnectionStrings.ts]: image

image

We good.

**Pull request merged**